apps/src/javalab/CommitDialog.jsx

```diff
@@ -115,16 +165,27 @@ export class UnconnectedCommitDialog extends React.Component {
   handleBackpackSaveSuccess = () => {
     const canClose =
       !this.state.commitSaveInProgress && !this.state.hasCommitSaveError;
-    this.setState({
-      hasBackpackSaveError: false,
-      backpackSaveInProgress: false,
-      filesToBackpack: []
-    });
+
+    this.props.backpackApi.getFileList(
+      this.handleBackpackLoadError,
+      filenames => {
+        this.setState({
+          hasBackpackSaveError: false,
+          hasBackpackLoadError: false,
+          backpackSaveInProgress: false,
+          filesToBackpack: [],
+          existingBackpackFiles: filenames
+        });
+      }
+    );
```

Comment:
Hmm...I'm wondering if it would be better to keep the previous state update to clear errors and saving in progress state before calling the backpack API? That way, if there's an error loading files, it might be a bit clearer that the file did successfully save, but there was a separate server error reloading the files? Not sure though, because it would also mean back to back state updates.

Reply:
Great point, check out the updated version and let me know what you think.